pkg/proxy/ipvs/proxier.go

```diff
@@ -718,30 +710,18 @@ func cleanupIptablesLeftovers(ipt utiliptables.Interface) (encounteredError bool
 	}
 
 	// Flush and remove all of our chains.
-	iptablesData := bytes.NewBuffer(nil)
-	if err := ipt.SaveInto(utiliptables.TableNAT, iptablesData); err != nil {
-		glog.Errorf("Failed to execute iptables-save for %s: %v", utiliptables.TableNAT, err)
-		encounteredError = true
-	} else {
-		existingNATChains := utiliptables.GetChainLines(utiliptables.TableNAT, iptablesData.Bytes())
-		natChains := bytes.NewBuffer(nil)
-		natRules := bytes.NewBuffer(nil)
-		writeLine(natChains, "*nat")
-		// Start with chains we know we need to remove.
-		for _, chain := range []utiliptables.Chain{kubeServicesChain, kubePostroutingChain, KubeMarkMasqChain} {
-			if _, found := existingNATChains[chain]; found {
-				chainString := string(chain)
-				writeLine(natChains, existingNATChains[chain]) // flush
-				writeLine(natRules, "-X", chainString)         // delete
+	for _, chain := range []utiliptables.Chain{kubeServicesChain, kubePostroutingChain, KubeMarkMasqChain} {
+		if err := ipt.FlushChain(utiliptables.TableNAT, chain); err != nil {
```

Comment:
This is a significant behavioral change - what was a single atomic op now becomes 2*N discrete operations, each of which will probably take longer to complete, and may not succeed because of ordering and link-counting (can't delete a chain that has inbound links from other chains).

What is the goal here?

Reply:
I think I missed that this was in ipvs, not iptables.  That makes it simpler.

@m1093782566 what do you think?

Reply:
If it can clean up legacy iptables rules well, great.

Reply:
Will take a deep look...

Reply:
Take another look and test it manually, it can clean up iptables rules. 

So, the changes make sense. Thanks!